dartagnan/src/main/java/com/dat3m/dartagnan/wmm/relation/Relation.java

```diff
@@ -183,6 +183,12 @@ public final BooleanFormula getSMTVar(Event e1, Event e2, SolverContext ctx) {
         return getSMTVar(new Tuple(e1, e2), ctx);
     }
 
+    public BooleanFormula getSMTCycleVar(Tuple edge, SolverContext ctx) {
+        return !getMaxTupleSet().contains(edge) ?
+        		ctx.getFormulaManager().getBooleanFormulaManager().makeFalse() :
+                edge(getName() + "-cycle", edge.getFirst(), edge.getSecond(), ctx);
+    }
+
```

Comment:
Isn't this just `getSMTVar` with a slightly different name?

Reply:
Yes, but the name is not a parameter, it is extracted from the `Relation` object.